stock_picking_operation_quick_change/wizards/stock_picking_wizard.py

```diff
@@ -0,0 +1,129 @@
+            "location_dest_id": picking.location_dest_id.id,
+        }
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        active_model = self.env.context["active_model"]
+        active_ids = self.env.context["active_ids"] or []
+        picking = self.env[active_model].browse(active_ids)
+        res.update(self._prepare_default_values(picking))
+        return res
+
+    def _default_old_dest_location_id(self):
+        stock_picking_obj = self.env["stock.picking"]
+        pickings = stock_picking_obj.browse(self.env.context["active_ids"])
+        first_move_line = pickings.mapped("move_line_ids")[:1]
+        return first_move_line.location_dest_id.id
+
+    def _get_allowed_locations(self):
+        return ["internal"]
+
+    def _get_allowed_location_domain(self):
+        return [("usage", "in", self._get_allowed_locations())]
+
+    def _get_allowed_picking_states(self):
+        return ["assigned"]
+
+    picking_id = fields.Many2one(comodel_name="stock.picking")
+    location_dest_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Actual destination location",
+        required=True,
+        readonly=True,
+    )
+    old_location_dest_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Old destination location",
+        default=_default_old_dest_location_id,
+        domain=lambda self: self._get_allowed_location_domain(),
+    )
+    new_location_dest_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="New destination location",
+        required=True,
+        domain=lambda self: self._get_allowed_location_domain(),
+    )
+    allowed_product_ids = fields.Many2many(
+        comodel_name="product.product",
+        compute="_compute_allowed_product_ids",
+    )
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        domain="[('id', 'in', allowed_product_ids)]",
+    )
+    change_all = fields.Boolean(
+        string="Change All",
+        help="Check if you want change all operations without filter "
+        "by old location",
+    )
+
+    @api.depends("picking_id")
+    def _compute_allowed_product_ids(self):
+        for record in self:
+            record.allowed_product_ids = record.picking_id.move_lines.mapped(
+                "product_id"
+            )
+
+    def check_allowed_pickings(self, pickings):
+        forbidden_pickings = pickings.filtered(
+            lambda x: x.state not in self._get_allowed_picking_states()
+        )
+        if forbidden_pickings:
+            raise UserError(
+                _(
+                    "You can not change operations destination location if "
+                    "picking state is not in %s"
+                )
+                % ",".join(self._get_allowed_picking_states())
+            )
+        pickings_with_chained_moves = pickings.filtered(
+            lambda x: x.move_lines.mapped("move_dest_ids")
```

Comment:
No need to use mapped on relational fields, Odoo does it by default
```suggestion
            lambda x: x.move_lines.move_dest_ids
```